.github/workflows/tests.yml

```
@@ -0,0 +1,49 @@
+
+      - name: Run golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: latest
+
+      - name: Run tests
+        run: go test -v -race -coverprofile=coverage.txt -covermode=atomic ./...
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v4
+if: matrix.go-version == '1.22'        with:
```

actionlint (1.7.10)

[error] could not parse as YAML: mapping values are not allowed in this context

(syntax-check)